domain/secretbackend/service/service_test.go

```diff
@@ -1219,6 +1219,43 @@ func (s *serviceSuite) TestSetModelSecretBackendIAASAuto(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 }
 
+func (s *serviceSuite) TestWatchModelSecretBackendChanged(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	svc := newWatchableService(
+		s.mockState, s.logger, s.mockWatcherFactory, s.clock,
+		func(backendType string) (provider.SecretBackendProvider, error) {
+			return providerWithConfig{
+				SecretBackendProvider: s.mockRegistry,
+			}, nil
+		},
+	)
+	modelUUID := coremodel.UUID(jujutesting.ModelTag.Id())
+	ch := make(chan struct{}, 1)
```

Comment:
Does this need to be buffered?
Also, should we send 2 events to check for the initial change and then a subsequent event?